### PR TITLE
FIx Logic for setFPColor and getFP Color

### PR DESCRIPTION
**Pull request by akhilbhas**
fixing following issues.
1) Index for the color read was wrong. - corrected indexing.
2) Color was read as int32 even though it is provided as string in YAML. - created a method to convert string  to the color value.

### CODE REVIEW

File: src/test_l1_dsFPD.c [hunk 1]
```diff
@@ -123,6 +120,43 @@ static bool extendedEnumsSupported=false; //Default to not supported
...
+
+
+
+static dsFPDColor_t getColorFromString(char* colorName)
+{
+    int count = 0;
+    dsFPDColor_t color = dsFPD_COLOR_WHITE;
+    for (count = 0; count < ((sizeof(dsFPColorMap))/sizeof(dsFPColorMap[0])); count ++)
+    {
+        if(!strncpy(colorName,dsFPColorMap[count].name,DS_FPD_KEY_SIZE)){
```
Comment (Anbukannadhasan) [inline src/test_l1_dsFPD.c:152]: Whether strncmp has to be used?

**Review by Anbukannadhasan — COMMENTED** [review]
added review comments

File: src/test_l1_dsFPD.c [hunk 2]
```diff
@@ -1198,38 +1233,30 @@ void test_l1_dsFPD_positive_dsSetFPColor (void)
     UT_ASSERT_EQUAL_FATAL(result, dsERR_NONE);
 
     // Step 02: Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()
-    // Step 03: Call dsSetFPColor() with all valid eIndicator and all valid eColor parameters from yaml
     count = UT_KVP_PROFILE_GET_UINT8("dsFPD/Number_of_Indicators");
```
Comment (Anbukannadhasan) [inline src/test_l1_dsFPD.c:1236]: Number_of_Indicators: 1. Please add this in Source_FPD.yaml. It is missing and it will cause parsing problem

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: src/test_l1_dsFPD.c [hunk 3]
```diff
@@ -1301,57 +1317,51 @@ void test_l1_dsFPD_negative_dsSetFPColor (void)
...
-            int isSupported = 0;
+            bool isSupported = false;
             for (int k = 1; k < numOfSupportedColors; ++k)
             {
-                snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/supportedColors/%d", eIndicator, j);
-                color =(dsFPDColor_t)UT_KVP_PROFILE_GET_UINT32(buffer);
-                if (allColors[j] == color)
+                snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/supportedColors/%d", i, k);
+                UT_KVP_PROFILE_GET_STRING(buffer,colorName);
+                if (!strncpy(colorName,dsFPColorMap[j].name,DS_FPD_KEY_SIZE))
```
Comment (Anbukannadhasan) [inline src/test_l1_dsFPD.c:1337]: Need to use strncmp?

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: src/test_l1_dsFPD.c [hunk 4]
```diff
@@ -1301,57 +1317,51 @@ void test_l1_dsFPD_negative_dsSetFPColor (void)
...
-	snprintf(supportedColorbuffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/supportedColors",eIndicator);
+        snprintf(supportedColorbuffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/supportedColors",i);
         numOfSupportedColors =UT_KVP_PROFILE_GET_LIST_COUNT(supportedColorbuffer);
 
-        for (int j = 0; j < sizeof(allColors)/sizeof(allColors[0]); ++j)
+        for (int j = 0; j < sizeof(dsFPColorMap)/sizeof(dsFPColorMap[0]); ++j)
         {
-            int isSupported = 0;
+            bool isSupported = false;
             for (int k = 1; k < numOfSupportedColors; ++k)
```
Comment (Anbukannadhasan) [inline src/test_l1_dsFPD.c:1333]: This loop should start with 0

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: src/test_l1_dsFPD.c (hunk 1 above)
Comment (akhilbhas) [reply src/test_l1_dsFPD.c]: Removed the function itself

File: src/test_l1_dsFPD.c (hunk 2 above)
Comment (akhilbhas) [reply src/test_l1_dsFPD.c:1203]: Added to source YAML

File: src/test_l1_dsFPD.c (hunk 3 above)
Comment (akhilbhas) [reply src/test_l1_dsFPD.c]: removed this logic

File: src/test_l1_dsFPD.c (hunk 4 above)
Comment (akhilbhas) [reply src/test_l1_dsFPD.c:1308]: updated

File: src/test_l2_dsFPD.c [hunk 5]
```diff
@@ -149,9 +179,9 @@ void test_l2_dsFPD_SetFPstateON_SetBrightness(void)
         }
 
         // Set the brightness of the LED
-        snprintf(minbuffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/MIN_BRIGHTNESS",eIndicator);
+        snprintf(minbuffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/MIN_BRIGHTNESS",i);
```
[the remark this answers is not in the document]
Comment (akhilbhas) [reply src/test_l2_dsFPD.c:152]: updated.

**Review by Anbukannadhasan — DISMISSED** [review]
Approved

File: src/test_l1_dsFPD.c [hunk 6]
```diff
@@ -1301,57 +1291,51 @@ void test_l1_dsFPD_negative_dsSetFPColor (void)
...
+                    isSupported = true;
                     break;
                 }
 
             }
-            if (!isSupported)
-            {
-                result = dsSetFPColor(eIndicator, allColors[j]);
+            if (!isSupported){
+                result = dsSetFPColor(eIndicator, dsFPAvailableColors[j]);
```
Comment (ishthiyaqahmed) [inline src/test_l1_dsFPD.c:1319]: please check what dsSetFPColor will return if we try to set the unsupported color? 
as per the interface file it is dsERR_OPERATION_NOT_SUPPORTED .

File: profiles/source/Source_FPD.yaml [hunk 7]
```diff
@@ -13,10 +14,10 @@ dsFPD:
       DEFAULT_COLOR_MODE : 0          # Color Mode of LED (Single(0) or Multicolored(1))
       MULTI_COLOR_MODE : 2
```
Comment (ishthiyaqahmed) [inline profiles/source/Source_FPD.yaml:15]: dont need multi color mode , decision can be made based on the default color mode .

File: src/test_l1_dsFPD.c (hunk 6 above)
Comment (akhilbhas) [reply src/test_l1_dsFPD.c]: updated

File: profiles/source/Source_FPD.yaml (hunk 7 above)
Comment (akhilbhas) [reply profiles/source/Source_FPD.yaml]: removed

File: profiles/source/Source_FPD.yaml [hunk 8]
```diff
@@ -11,8 +11,7 @@ dsFPD:
       MIN_BRIGHTNESS : 0              # Minimum brightness value of FPD LEDs
       DEFAULT_LEVELS : 1              # Default Brightness level for the FP text display
       MAX_CYCLERATE : 2               # Maximum Rate at which LED is rotating during scrolling
-      DEFAULT_COLOR_MODE : 0          # Color Mode of LED (Single(0) or Multicolored(1))
-      MULTI_COLOR_MODE : 2
+      DEFAULT_COLOR_MODE : 1          # Color Mode of LED (Single(0) or Multicolored(1))
       supportedColors :
```
Comment (Anbukannadhasan) [inline profiles/source/Source_FPD.yaml:15]: Can you please add comments as present in the videoport yaml files

example:
#dsFPD_COLOR_BLUE   dsFPDColor_Make(0, 0, 0xFF)          ///< Blue color LED                 #
#dsFPD_COLOR_GREEN ///< Green color LED        
#dsFPD_COLOR_RED     ///< Red color LED                 
#dsFPD_COLOR_YELLOW  ///< Yellow color LED               
#dsFPD_COLOR_ORANGE  ///< Orange color LED               
#dsFPD_COLOR_WHITE   ///< White color LED       
      supportedColors :
      - dsFPD_COLOR_GREEN
      - dsFPD_COLOR_RED
      - dsFPD_COLOR_ORANGE
      - dsFPD_COLOR_WHITE

Please consider for both source and sink yaml. Also consider for the Indicator_Type ,  SupportedLEDStates

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comment

**Review by ishthiyaqahmed — APPROVED** [review]

**Review by srinivasgtl — APPROVED** [review]
Approved

**Review by hari22yuva — APPROVED** [review]
Approved

**Pull request merged**